### PR TITLE
Enable automatic `logo` insertion

### PR DESCRIPTION
**Pull request by huong-li-nguyen**
## Description
- Enable automatic `logo` insertion into the `page-header` container - this means that the first file found called `logo. xxx` in the assets folder will be positioned in the left corner
- The `settings` container will now move to the `right-header` container if both `dashboard-title` and `logo` are hidden
- Note: There might always be some custom CSS required to position the logo nicely because a logo can have different paddings, which we cannot predict beforehand. The best thing we can do is to put no restriction on the logo image and always let it take 100% of the height of its parent container. Repositioning can then be done via custom CSS  (see code changes in the examples app)

@nadijagraca  - thanks for creating the first POC on this! I've had to update it based on the new container structure and logic we will likely introduce [here](https://github.com/mckinsey/vizro/pull/238). Depending on your capacity you could help me finish this PR off with docs and tests :) 

TO DO in separate PRs:

- [ ] docs + tests

### Logo ⛔ + Dashboard Title ✅ 
no_logo_title

### Logo ✅  + Dashboard Title ✅ 
logo_title

### Logo ✅  + Dashboard Title ⛔
logo_no_title

### Logo ⛔ + Dashboard Title ⛔
no_logo_no_title

### Rectangular Logo ✅  + Dashboard Title ⛔
See this commit to see the relevant CSS changes: https://github.com/mckinsey/vizro/pull/248/commits/d71f101488bfe7d23b06b1ceaa87fee7c6cf3a89

Screenshot 2024-01-04 at 14 16 56

## Notice

- [x] I acknowledge and agree that by checking this box and clicking "Submit Pull Request,":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate anyone else's intellectual property rights.
  - I have not referenced individuals, products, or companies in any commitments, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/src/vizro/static/css/layout.css [hunk 1]
```diff
@@ -182,10 +183,23 @@ div.dashboard_container .custom-tooltip {
...
+
+#logo {
+  height: 100%;
+}
+
+#logo-and-title {
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+  gap: 12px;
```
Comment (nadijagraca) [inline vizro-core/src/vizro/static/css/layout.css:203]: Should we maybe increase the gap between the logo and the title? It looks a bit off as the logo does not align with the nav bar below. 

Screenshot 2024-01-16 at 15 28 48

Increasing it to 16px leads to better alignment with the navbar. 
Screenshot 2024-01-16 at 15 27 46

**Review by nadijagraca — APPROVED** [review]
LGTM. I just have one small comment on css.

File: vizro-core/src/vizro/static/css/layout.css (hunk 1 above)
Comment (huong-li-nguyen) [reply vizro-core/src/vizro/static/css/layout.css:203]: I like that proposal! Will implement :)

File: vizro-core/src/vizro/models/_dashboard.py [hunk 2]
```diff
@@ -144,6 +144,9 @@ def _get_page_divs(self, page: Page) -> _PageDivsType:
             ),
             id="settings",
         )
+        logo_img = self._infer_image(filename="logo")
```
Comment (antonymilne) [inline vizro-core/src/vizro/models/_dashboard.py:147]: Please could you make this `_infer_image` a `@staticmethod` while I remember?

**Review by antonymilne — APPROVED** [review]
🌟 Looks great!

**Pull request merged**